forte/data/data_pack.py

```diff
@@ -1597,28 +1619,58 @@ def require_annotations(entry_class=Annotation) -> bool:
             yield from []
             return
 
+        # If range_annotation is specified, we record its begin and
+        # end index
+        range_begin: int
+        range_end: int
+
+        if range_annotation is not None:
+            if isinstance(range_annotation, (Annotation, AudioAnnotation)):
```

Comment:
We might want to create a type of joint type for `AnnotationLike` so we don't have to hard-code.